src/uwtools/config/tools.py

```diff
@@ -127,11 +127,12 @@ def _ensure_format(
     """
     if isinstance(config, Config):
         return config.get_format()
+    if isinstance(config, Path):
+        return fmt or get_file_format(config)
+    if isinstance(config, dict):
+        return fmt or FORMAT.yaml
```

Comment:
Here's another case where, if we are operating on a plain `dict`, we associate it with the YAML format for convenience.